astropy/table/table.py

```diff
@@ -16,6 +16,7 @@
 
 from .. import log
 from ..io import registry as io_registry
+from ..io.fits.util import  _is_int
```

Comment:
We should not be importing a private function from another module!
In principle, python has the `__index__` method to check whether something can be used as an index, so you could write
```
def _is_int(item):
    try:
        item.__index__()
    except Exception:
        return False
    else:
        return True
```
This works for numpy integers as well.

I would probably put this function in `utils.misc as `isint`